homeassistant/components/calendar/__init__.py

```diff
@@ -386,6 +386,8 @@ async def get(self, request: web.Request, entity_id: str) -> web.Response:
             return web.Response(status=HTTPStatus.BAD_REQUEST)
         if start_date is None or end_date is None:
             return web.Response(status=HTTPStatus.BAD_REQUEST)
+        if start_date > end_date:
+            return web.Response(status=HTTPStatus.BAD_REQUEST)
```

Comment:
This is a Core entity component, considering its central role, please add tests for this case.

../Frenck